Sources/AppStoreConnectCLI/Services/Operations/ListBuildsOperation.swift

```diff
@@ -36,21 +29,38 @@ struct ListBuildsOperation: APIOperation {
         filters += options.filterProcessingStates.isEmpty ? [] : [.processingState(options.filterProcessingStates)]
         filters += options.filterBetaReviewStates.isEmpty ? [] :  [.betaAppReviewSubmissionBetaReviewState(options.filterBetaReviewStates)]
 
-        let limit = options.limit.map { limit -> [ListBuilds.Limit] in
+        return filters
+    }
+
+    var limit: [ListBuilds.Limit]? {
+        options.limit.map { limit -> [ListBuilds.Limit] in
             [.individualTesters(limit), .betaBuildLocalizations(limit)]
         }
+    }
+
+    private let options: Options
 
-        let endpoint = APIEndpoint.builds(
-            filter: filters,
-            include: [.app, .betaAppReviewSubmission, .buildBetaDetail, .preReleaseVersion],
-            limit: limit,
-            sort: [ListBuilds.Sort.uploadedDateAscending]
-        )
+    init(options: Options) {
+        self.options = options
+    }
 
-        return requestor.request(endpoint)
-            .map { response -> Output in
-                response.data.map { ($0, response.included) }
+    func execute(with requestor: EndpointRequestor) -> AnyPublisher<Output, Error> {
+        requestor.requestAllPages {
+            .builds(
+                filter: self.filters,
+                include: [.app, .betaAppReviewSubmission, .buildBetaDetail, .preReleaseVersion],
+                limit: self.limit,
+                sort: [.uploadedDateAscending],
+                next: $0
+            )
+        }
+        .map { (responses: [BuildsResponse]) -> Output in
+            responses.flatMap { (response: BuildsResponse) -> Output in
+                (response.data.map { ($0, response.included) })
             }
-            .eraseToAnyPublisher()
+        }
+        .eraseToAnyPublisher()
```

Comment:
Might be more readable if we pull out the endpoint params, then we can put the endpoint provider function on one line

```swift
let filter = self.filters
let include: [...] = [.app, .betaAppReviewSubmission, .buildBetaDetail, .preReleaseVersion]
let limit = self.limit
let sort: [...] = [.uploadedDateAscending]

return requestor
    .requestAllPages { .builds(filter: filters, include: includes, limit: limit, sort: sort, next: $0) }
    .map { (responses: [BuildsResponse]) -> Output in
        responses.flatMap { (response: BuildsResponse) -> Output in
            (response.data.map { ($0, response.included) })
        }
    }
    .eraseToAnyPublisher()
```